### DISCUSSION TITLE
sub mod for Cosplay Roleplay

### DISCUSSION THREAD

**Discussion by howltrek** [Monika-After-Story/MonikaModDev#9380] [general - submod / external content]
https://www.mediafire.com/file/9ywnh5ik1j4bex9/Mony+succubus+Role+Play+Mod+beta.zip/file

this is a submod, which i have been working on.

The idea is that Monika plays some character, in a new category of submods.
I beg you to try it and share your opinions.

![screenshot0102](https://user-images.githubusercontent.com/90975566/180351547-f78705d1-6609-409d-ae43-5ef105a9e457.png)
![screenshot0103](https://user-images.githubusercontent.com/90975566/180351534-cc15ca54-0560-4456-a934-6ced53f3a5d8.png)
![screenshot0104](https://user-images.githubusercontent.com/90975566/180351539-77a8232b-b222-4e59-8554-7fa18a4e3930.png)
![screenshot0105](https://user-images.githubusercontent.com/90975566/180351541-b97aa176-5940-4f8d-a5df-57b2c8d5bc57.png)

**Comment by howltrek** [comment]
I made a few small arrangements and put some lines that I had forgotten to add

**Reply by howltrek** [reply]
what do you think of the concept?

![succubus full-Recuperado](https://user-images.githubusercontent.com/90975566/184561626-e9e529cc-6f86-4dec-9b29-74a15b28b29b.png)

**Reply by Delcos** [reply]
Looks great, this would be a pretty cool submod especially if it goes further, a lot of potential.

**Reply by Justformas** [reply]
Yeah agree with Delcos. You going to incorporate it into this submod somehow?

**Comment by Justformas** [comment]
This is pretty cool. Even has a few different cosplay scripts to go through, and I like that you can revisit any one you want after going through them all.

In a line or few it sounds like Monika was saying lines that the player would say, which was kind of confusing. Maybe if "Player:" or "(player's name):" was present during those lines it'd be more clear. Examples: "Wait a minute! were you bewitching me?" "How did you plan to do that?" "So if you kiss me, will you leave this place forever?" "In that case, I'd rather refuse."

Also I think at one point, she says "player" rather than the player's name/nickname: "Sure player, I have something special planned for today."

Just so you know it overwrites the files from the other succubus costume you had made. It looks to be the same costume, but I re-gifted anyway. I like the new dialogue she says when selecting it too, and having poems that were there in the cosplay script.

Instead of her changing back into her default school uniform clothing afterwards, is there a way to have her put back on the last clothing she was wearing? Or maybe she could ask if you want her to keep the succubus costume on at the end of every script (it does this for one script but not the others).

**Comment by howltrek** [comment]
> This is pretty cool. Even has a few different cosplay scripts to go through, and I like that you can revisit any one you want after going through them all.
> 
> In a line or few it sounds like Monika was saying lines that the player would say, which was kind of confusing. Maybe if "Player:" or "(player's name):" was present during those lines it'd be more clear. Examples: "Wait a minute! were you bewitching me?" "How did you plan to do that?" "So if you kiss me, will you leave this place forever?" "In that case, I'd rather refuse."
> 
> Also I think at one point, she says "player" rather than the player's name/nickname: "Sure player, I have something special planned for today."
> 
> Just so you know it overwrites the files from the other succubus costume you had made. It looks to be the same costume, but I re-gifted anyway. I like the new dialogue she says when selecting it too, and having poems that were there in the cosplay script.
> 
> Instead of her changing back into her default school uniform clothing afterwards, is there a way to have her put back on the last clothing she was wearing? Or maybe she could ask if you want her to keep the succubus costume on at the end of every script (it does this for one script but not the others).

Thanks for the review.
I think I fixed all the problems you mentioned.
(Except for Monika sometimes calling the player by name. That was intentional, to indicate that Monika was out of character as the succubus.)

If you have time, I'd appreciate it if you could download it again and try it out.

**Comment by Justformas** [comment]
> If you have time, I'd appreciate it if you could download it again and try it out.

Done, so now the player says their lines themselves through a menu. That's more clear and feels more immersive.

> I think I fixed all the problems you mentioned.
> (Except for Monika sometimes calling the player by name. That was intentional, to indicate that Monika was out of character as the succubus.)

I think you fixed what I was referring to, that she literally said "player" instead of the player's name/nickname at one point, but seems to be fixed now.

So does she only ask if you want her to keep the costume on after the 4th script because that's the last one? For the others, she doesn't give that option and just changes back to her default clothing. I guess it's not a huge deal to ask her to change her clothes again, and players may just go through all 4 scripts at once anyway. I'm not sure if there's a way for her to remember the last clothes she had on and change back into those either.

By the way, I'm not sure if you care about fixing grammar stuff right now. I figured it might not be priority right now, and didn't want to try documenting every example.

**Comment by howltrek** [comment]
Justformas

I added the option to ask Monica to wear the costume, in the final menu.
Unfortunately, for the grammar problem, I would need collaboration, since I am a Spanish speaker, so mine is very mediocre.
![Captura de pantalla (122)](https://user-images.githubusercontent.com/90975566/180588465-64dbc79e-ca33-4354-95da-686792e06985.png)

**Comment by Justformas** [comment]
> I added the option to ask Monica to wear the costume, in the final menu.

Ok that helps, and also makes sense to have that in that menu.

> Unfortunately, for the grammar problem, I would need collaboration, since I am a Spanish speaker, so mine is very mediocre.

I heard you say something like that before, but in these messages your grammar seems good.

A lot of the things I noticed are actually little things like the first letter of a line not being capitalized, or missing periods at the end (though sometimes this stuff might be intentional), or the word "I" not being capitalized, and maybe some comma things.

Not much else that stood out:

`"I love the sensuality with Emily Dickinson expresses in this poem."` The word "with" should be replaced with "that".
`"I'm sorry for I don't have anything new."` The word "for" should be replaced with "that".

Interestingly, I guess "Jessamines" is another word for "Jasmines", and both have been used in different versions of that poem. Didn't know.

**Reply by howltrek** [reply]
corrections made

**Comment by Delcos** [comment]
I may just be an idiot, but I've installed this and she just says she doesn't recognize the gift. Sorry if I'm missing something obvious.

**Reply by howltrek** [reply]
I don't know what you did

**Reply by Justformas** [reply]
You have to copy the "mod_assets" folder into the "game" folder of your MAS directory first for her to recognize the gift. Also for this Submod to work you need to copy the "Submods" folder into the "game" folder as well. If it asks if you want to merge folders then say yes. Try giving her the gift after you do those things.

For installing spritepacks in general there's a guide here: https://github.com/Monika-After-Story/MonikaModDev/wiki/Adding-Sprite-Objects

**Reply by Delcos** [reply]
> I don't know what you did

So she accepted it now, but what is the trigger. I'm guessing maybe a set time of night but is there any conditions that need to be met first.

**Reply by howltrek** [reply]
the submod will only be visible to players whose Monika is ENAMORED or more.
I didn't want the mod to steal the first kiss with Monika from some player.

**Reply by multimokia** [reply]
As an FYI ,your submod still breaks flow as enamored doesn't necessarily mean had first kiss. It just means eligible. Nowhere in your submod do you perform the check.

**Reply by howltrek** [reply]
It was what worried me the most and I thought that that was enough, now I don't know how to fix it.

**Reply by Justformas** [reply]
Would it work to add in the addEvent line `conditional="persistent._mas_first_kiss is not None"`
and an action tie the condition to it, in this case the unlocking action so `action=EV_ACTION_UNLOCK`?
Then you might not need the enamored check either.

**Reply by howltrek** [reply]
after a long time, i managed to make a script that checks if the first kiss has happened.
Tomorrow I will implement it in the submod


init 5 python:
    addEvent(Event(persistent.event_database,eventlabel="test",unlocked=True,category=['appearance','clothes','roleplay'],prompt="test"))
label test: 
    python:
        notimeforlove= persistent._mas_first_kiss is None

    if notimeforlove is True:
        m "not now"
    else:
        call monika_kissing_motion_short
        m "is working"
return

**Reply by howltrek** [reply]
> As an FYI ,your submod still breaks flow as enamored doesn't necessarily mean had first kiss. It just means eligible. Nowhere in your submod do you perform the check.

I make the relevant changes, I hope it works.

**Reply by Justformas** [reply]
So you added dialogue for that and also for affection levels lower than enamored. But you still have `aff_range=(mas_aff.ENAMORED, None)))` in the event block, so wouldn't the player still have to be enamored or higher to see this topic in the first place? Also one thing about this line: `"The costume is too lewd and the script too embarrassing for me..."`

She says the costume is too lewd if your affection range is higher than upset but less than enamored, but technically as long as the player has at least 30 affection, can't they get Monika to wear the costume through the regular "Can you change your clothes?" option? They wouldn't be able to do the script, but then it'd seem that it's the script specifically that Monika wouldn't be comfortable doing, rather than wearing the costume.

**Reply by howltrek** [reply]
I had deleted it. but I must work on an older version of the script by mistake.
I'll correct it now, the suit thing will take a while.

I have in mind a somewhat improved version of this, which will only be accessible with the submod and when I finish it, I will make the correct .json for it.

**Reply by Justformas** [reply]
Well originally when this wasn't a submod and just a Succubus costume, I did like that the player could have Monika wear it with just 30 affection, or not long after starting the mod. Maybe that version can still exist, but with this submod, it makes sense to restrict it if that's what she's going to say.

**Reply by howltrek** [reply]
you're right, a sprite pack like this sounds weird, for now I'll just remove the dialogue about the lewd of the costume

**Reply by Justformas** [reply]
By the way, since you removed the first part of that sentence that had the word "is" in it, it should be added in the remaining part so that it's "The script is too embarrassing for me...". But I think this submod is coming along really well.

**Comment by Justformas** [comment]
Ok new topic(s). I'm not sure if you care about this (I don't think it's a big deal), but for the option you added to ask Monika to just wear the costume, if she already has the costume on, she still acts like she doesn't have it on. So in that case, while you could maybe have it check for that and have her say something like "But I'm already wearing the costume!", it's not a big deal.

Also this is more about "hover_dlg" in the json, but how am I supposed to get it to trigger in game? I try to hover over the succubus costume icon after selecting it, but it doesn't work. Nor does it work when I hover over Monika herself after selecting it. I'm wondering if it's implemented correctly or if I'm just not doing it right.

**Reply by howltrek** [reply]
I'm going to review that

**Reply by howltrek** [reply]
I already made the modification.
I had a hard time figuring out how to do it.

**Reply by howltrek** [reply]
I forgot to mention.
it is necessary to replace even the costume, for it to work, since the mod reads a label on it, to know if Monika is wearing it or not

**Reply by Justformas** [reply]
Oh you added something for her to say when she's already in the costume. I like it.

But I still can't figure out how to get the hover dialogue to trigger in game. Have you tested it and gotten it to trigger? What exactly are you supposed to do to trigger it? I'm wondering if this is related to:
https://github.com/Monika-After-Story/MonikaModDev/wiki/Adding-Sprite-Objects#version-history

"0.11.0 - added keep_on_desk property to ACSTemplates. Removed hover_dlg from JSONs."

**Comment by Polariusz** [comment]
Slight issue, in line 298; m 5nsbsb "Sure **[mas_get_player_nickname()**, I have something special planned for today."
**[mas_get_player_nickname()** isn't closed, just add the ']' at the end of it.

**Reply by howltrek** [reply]
done

**Comment by M1ntyX0X** [comment]
with me she says that she would be uncomfortable, does this mean that there's like an affection lock for when you can activate it or something?

**Reply by howltrek** [reply]
She need he first kiss

**Reply by M1ntyX0X** [reply]
oh kk

**Reply by M1ntyX0X** [reply]
thanks

**Comment by Delcos** [comment]
Any new release links?

**Comment by Justformas** [comment]
Was going to ask if more was going to be added to this, but noticed something else. I think this can actually break flow for mas_lingerie_intro which is an event that triggers the first time Monika wears lingerie. But it only triggers for Christmas or Valentine's Day lingerie, since normally one of those would be the first. But also, it won't happen if mas_hasUnlockedClothesWithExprop('lingerie') is true.

I think gifted clothing with jsons labeled as lingerie like the Succubus costume cause that to be true. Otherwise I don't know how I got the Devil's Night/Halloween lingerie, which also relies on that being true, and the first time lingerie intro can't happen there. I haven't gone through Christmas or Valentine's Day yet. But I don't really mind that I got this first. Didn't even know I wouldn't have otherwise.
